packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -730,3 +715,738 @@ class _RenderChildOverflowBox extends RenderAligningShiftedBox {
+  static const dynamic titleSmallWeight = _TokenTypeface.weightMedium;
+}
+
+
+// ----------------------------------------------------------------------
+// The following could also be generated or included in the templates
+// for color and typefaces as they are needed to translate the base
+// tokens into Colors and TypeStyles.
+//
+enum _ColorType {
+  primary,
+  onPrimary,
+  primaryContainer,
+  onPrimaryContainer,
+  secondary,
+  onSecondary,
+  secondaryContainer,
+  onSecondaryContainer,
+  tertiary,
+  onTertiary,
+  tertiaryContainer,
+  onTertiaryContainer,
+  error,
+  onError,
+  errorContainer,
+  onErrorContainer,
+  outline,
+  background,
+  onBackground,
+  surface,
+  onSurface,
+  surfaceVariant,
+  onSurfaceVariant,
+  inverseSurface,
+  onInverseSurface,
+  inversePrimary,
+  shadow,
+}
+
+Color _colorForType(BuildContext context, _ColorType type) {
+  final ColorScheme colors = Theme.of(context).colorScheme;
+  switch (type) {
+    case _ColorType.primary: return colors.primary;
+    case _ColorType.onPrimary: return colors.onPrimary;
+    case _ColorType.primaryContainer: return colors.primaryContainer;
+    case _ColorType.onPrimaryContainer: return colors.onPrimaryContainer;
+    case _ColorType.secondary: return colors.secondary;
+    case _ColorType.onSecondary: return colors.onSecondary;
+    case _ColorType.secondaryContainer: return colors.secondaryContainer;
+    case _ColorType.onSecondaryContainer: return colors.onSecondaryContainer;
+    case _ColorType.tertiary: return colors.tertiary;
+    case _ColorType.onTertiary: return colors.onTertiary;
+    case _ColorType.tertiaryContainer: return colors.tertiaryContainer;
+    case _ColorType.onTertiaryContainer: return colors.onTertiaryContainer;
+    case _ColorType.error: return colors.error;
+    case _ColorType.onError: return colors.onError;
+    case _ColorType.errorContainer: return colors.errorContainer;
+    case _ColorType.onErrorContainer: return colors.onErrorContainer;
+    case _ColorType.outline: return colors.outline;
+    case _ColorType.background: return colors.background;
+    case _ColorType.onBackground: return colors.onBackground;
+    case _ColorType.surface: return colors.surface;
+    case _ColorType.onSurface: return colors.onSurface;
+    case _ColorType.surfaceVariant: return colors.surfaceVariant;
+    case _ColorType.onSurfaceVariant: return colors.onSurfaceVariant;
+    case _ColorType.inverseSurface: return colors.inverseSurface;
+    case _ColorType.onInverseSurface: return colors.onInverseSurface;
+    case _ColorType.inversePrimary: return colors.inversePrimary;
+    case _ColorType.shadow: return colors.shadow;
+  }
+}
+
+Color _resolveColor(BuildContext context, _ColorType type, [double? opacity]) {
+  final Color color = _colorForType(context, type);
+  return (opacity != null)
+      ? color.withOpacity(opacity)
+      : color;
+}
+
+TextStyle _textStyleForValue(_TokenTypeScaleValue value) {
```

Comment:
Once the M3 text styles are in, I imagine this can be simplified down to something like this:

```dart
TextStyle _textStyleForValue(TextTheme textTheme, _TokenTypeScaleValue value) {
  switch (value) {
    case _TokenTypeScaleValue.bodyLarge:
      return textTheme.bodyLarge;
  ...
}
```